### PR TITLE
fix: misc UI fixes around creation and config

### LINKED ISSUE #6769: Azure subscription list drop-down is not sorted alphabetically
## Describe the bug

List of Azure subscriptions is not filtered alphabetically 

## Version

1.4.0-nightly.232669

## Browser

- [x] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [ ] macOS
- [x] Windows
- [ ] Ubuntu

## To Reproduce

Select LUIS/QNA key, select Azure subscription and see the drop down list contents

## Expected behavior

Sorted A-Z enabling easy retrieval of the right sub

## Screenshots

image

### LINKED ISSUE #6815: New Bot Creation experience shouldn't use technology choices in user status messages
## Describe the bug
When the new bot creation experience starts doing the work to install the chosen template, running the template, etc, we display status to the user, which is good.  However, we also display the underlying technology we use for the templates, Yeoman, which isn't necessary.

See screenshots below

## Version
Version:  1.4.0-nightly.232669.4daad5c
Electron: 8.2.4
Chrome: 80.0.3987.165
NodeJS: 12.13.0
V8: 8.0.426.27-electron.0

## Browser
- [x] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS
- [x] macOS
- [ ] Windows
- [ ] Ubuntu

## To Reproduce

1. Launch Composer
2. Click the '+' to create a new bot
3. Select any bot template, empty or whatever we're calling basic bot these days is good enough
4. Fill in the required data
5. Once the creation experience starts to download the template, we display a spinner and provide status text of the current creation process.  See screenshots below
6. We shouldn't include the technology choice in the status message.  There isn't really a good reason to let the user know we are using Yeoman.

## Expected behavior
Remove "Yeoman" from the status text messages

## Screenshots
Screen Shot 2021-04-08 at 9 16 02 AM

Screen Shot 2021-04-08 at 9 16 08 AM

### LINKED ISSUE #6822: Selecting LUIS and QnA keys results in errors that settings aren't set (but are!)
## Describe the bug

After creating a conversational core bot, then following LUIS and QnA key selection prompts in the rocket ship UI, it dropped me into the page shown below with errors saying the luis region, qna key wasn't set. This appears to be a refresh issue as they are set, clicking away and back to settings resolves the issue

## Version

1.4.0-nightly.233315

## Browser

- [x] Electron distribution
- [ ] Chrome
- [ ] Safari
- [ ] Firefox
- [ ] Edge

## OS

- [x] macOS
- [ ] Windows
- [ ] Ubuntu

## To Reproduce

1. Create a Conversational Core project
2. Click LUIS and QNA rocket ship key selection UX and complete
3. Observe the next landing page has red errors

## Expected behavior

Once the rocket ship prompted LUIS and QnA settings are completed the page it drops you on should show those keys and not (incorrectly) show that they are missing?

## Screenshots

image

image

### LINKED ISSUE #6837: Redundant loading spinners when opening a bot
When you open a bot two spinners are used in the opening flow. This UI needs to be normalized to only use one.

### PR DESCRIPTION
**Pull request by pavolum**
## Description

This PR contains various fixes for UI bugs that were filed during initial bug bashing. More details below in the Task Item header.

## Task Item

fix #6769 - Added sorting to api call results for getSubscription and getResourceGroups
fix #6815 - Removed 'yeoman' string from status messages
fix #6837 - There have always been two loading spinners in the open project flow, one for bot opening and one for design view loading. They were previously identical so the end user saw no visible change in the loading UI when running the bot opening flow. Since then the botOpening loading spinner UI was updated. This PR updates the design page spinner UI to match the updated bot opening so end users see the same loading UI for both spinners. 
fix #6822 - in development

### CODE REVIEW

**Comment by pavolum** [comment]
@benbrown it seems there is a bit of redundant code in the qnaModal and LuisModal components when they call azure to get the users available subscriptions. I also see that code was copied from the azurePublish extension. Do we have an item tracking a minor refactor? It would be ideal to have a shared workspace where we could put this function and have it used by the client workspace and AzurePublish extension that way it is written once and used in three seperate places as opposed to the current situation where we have the same logic written in three different places.

**Comment by benbrown** [comment]
@pavolum  yes it is on the backlog to refactor these into a single shared component.

File: Composer/packages/client/src/components/LoadingSpinner.tsx [hunk 1]
```diff
@@ -15,15 +16,39 @@ const container = css`
...
   const { message } = props;
-  return (
-    <div css={container}>
-      <Spinner label={message || formatMessage('Loading')} />
-    </div>
+
+  const renderLoadingSpinner = () => {
+    return (
+      <div css={container}>
+        <Spinner label={message || formatMessage('Loading')} />
```
Comment (beyackle) [inline Composer/packages/client/src/components/LoadingSpinner.tsx:34]:
```suggestion
        <Spinner label={message ?? formatMessage('Loading')} />
```

File: Composer/packages/client/src/components/ManageLuis/ManageLuis.tsx [hunk 2]
```diff
@@ -83,7 +83,15 @@ export const ManageLuis = (props: ManageLuisProps) => {
...
-      return subscriptionsResult._response.parsedBody;
+      return subscriptionsResult._response.parsedBody.sort((a, b) => {
+        if (a?.displayName && b.displayName) {
+          const nameA = a.displayName.toUpperCase();
+          const nameB = b.displayName.toUpperCase();
+          return nameA < nameB ? -1 : nameA > nameB ? 1 : 0;
+        } else {
+          return 0;
+        }
+      });
```
Comment (beyackle) [inline Composer/packages/client/src/components/ManageLuis/ManageLuis.tsx:94]: `lodash.sortBy` can simplify this.

File: Composer/packages/client/src/components/ManageQNA/ManageQNA.tsx [hunk 3]
```diff
@@ -93,7 +93,15 @@ export const ManageQNA = (props: ManageQNAProps) => {
...
-      return subscriptionsResult._response.parsedBody;
+      return subscriptionsResult._response.parsedBody.sort((a, b) => {
+        if (a?.displayName && b.displayName) {
+          const nameA = a.displayName.toUpperCase();
+          const nameB = b.displayName.toUpperCase();
+          return nameA < nameB ? -1 : nameA > nameB ? 1 : 0;
+        } else {
+          return 0;
+        }
+      });
```
Comment (beyackle) [inline Composer/packages/client/src/components/ManageQNA/ManageQNA.tsx:104]: `lodash.sortBy` here too.

File: extensions/azurePublish/src/components/api.tsx [hunk 4]
```diff
@@ -45,7 +45,11 @@ export const getSubscriptions = async (token: string): Promise<Array<Subscriptio
       });
       throw new Error(subscriptionsResult._response.bodyAsText);
     }
-    return subscriptionsResult._response.parsedBody;
+    return subscriptionsResult._response.parsedBody.sort((a, b) => {
+      const nameA = a.displayName.toUpperCase();
+      const nameB = b.displayName.toUpperCase();
+      return nameA < nameB ? -1 : nameA > nameB ? 1 : 0;
+    });
```
Comment (beyackle) [inline extensions/azurePublish/src/components/api.tsx:52]: `lodash.sortBy`

File: extensions/azurePublish/src/components/api.tsx [hunk 5]
```diff
@@ -81,7 +85,11 @@ export const getResourceGroups = async (token: string, subscriptionId: string):
       });
       return [];
     }
-    return resourceGroupsResult._response.parsedBody;
+    return resourceGroupsResult._response.parsedBody.sort((a, b) => {
+      const nameA = a.name.toUpperCase();
+      const nameB = b.name.toUpperCase();
+      return nameA < nameB ? -1 : nameA > nameB ? 1 : 0;
+    });
```
Comment (beyackle) [inline extensions/azurePublish/src/components/api.tsx:92]: `lodash.sortBy`

**Review by beyackle — CHANGES_REQUESTED** [review]
Using lodash even for things you know how to write is better for unit-test coverage (because we don't need to worry about testing simple things like sorting, so those lines get left out of the denominator of the fraction).

File: Composer/packages/client/src/components/LoadingSpinner.tsx (hunk 1 above)
Comment (pavolum) [reply Composer/packages/client/src/components/LoadingSpinner.tsx]: Handled

File: Composer/packages/client/src/components/ManageLuis/ManageLuis.tsx (hunk 2 above)
Comment (pavolum) [reply Composer/packages/client/src/components/ManageLuis/ManageLuis.tsx]: Made the change

File: extensions/azurePublish/src/components/api.tsx (hunk 4 above)
Comment (pavolum) [reply extensions/azurePublish/src/components/api.tsx]: Made the change

File: extensions/azurePublish/src/components/api.tsx (hunk 5 above)
Comment (pavolum) [reply extensions/azurePublish/src/components/api.tsx]: Made the change

File: Composer/packages/client/src/components/ManageQNA/ManageQNA.tsx (hunk 3 above)
Comment (pavolum) [reply Composer/packages/client/src/components/ManageQNA/ManageQNA.tsx]: Made the change

**Review by beyackle — DISMISSED** [review]
LGTM.

**Review by benbrown — APPROVED** [review]

**Review by srinaath — APPROVED** [review]

**Review by beyackle — APPROVED** [review]

**Pull request merged**